src/com/owncloud/android/ui/preview/PreviewImageActivity.java

```diff
@@ -103,8 +105,10 @@ protected void onCreate(Bundle savedInstanceState) {
         actionBar.hide();
         
         // PIN CODE request
-        if (getIntent().getExtras() != null && savedInstanceState == null && fromNotification()) {
-            requestPinCode();
+        if (PinCheck.checkIfPinEntry()){
```

Comment:
We should refactor all this, so that FileActivity does the work with the PIN code, and not every of it's children. That would not cover all the activities, but would save some repeated code.